yaml_test.go

```diff
@@ -145,69 +161,107 @@ func TestUnmarshalNonStrict(t *testing.T) {
 		},
 	} {
 		s := UnmarshalString{}
-		unmarshalEqual(t, tc.yaml, &s, &tc.want)
+		unmarshalEqual(t, tc.yaml, &s, &tc.want, false)
 	}
 }
 
-// prettyFunctionName converts a slice of JSONOpt function pointers to a human
-// readable string representation.
-func prettyFunctionName(opts []JSONOpt) []string {
-	var r []string
-	for _, o := range opts {
-		r = append(r, runtime.FuncForPC(reflect.ValueOf(o).Pointer()).Name())
-	}
-	return r
-}
+func unmarshalEqual(t *testing.T, y []byte, s, e interface{}, knowFields bool) {
```

Comment:
This should probably be `knownFields`. I'm not a fan of boolean flags though, they're hard to read. Might it be better to define specific functions for each use-case? i.e.:

* `unmarshalEqual`
* `decodeEqual`
* `decodeEqualKnownFields`